addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,899 @@
+        // Nothing to do here
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (!(command instanceof RefreshType)) {
+            if (channelUID != null) {
+                Channel channel = this.getThing().getChannel(channelUID.getId());
+                if (channel != null) {
+                    Configuration channelConfiguration = channel.getConfiguration();
+                    if (channel.getAcceptedItemType().equals(IRtransBindingConstants.BLASTER_CHANNEL_TYPE)) {
+                        if (command instanceof StringType) {
+                            String remoteName = StringUtils.substringBefore(command.toString(), ",");
+                            String irCommandName = StringUtils.substringAfter(command.toString(), ",");
+
+                            IrCommand ircommand = new IrCommand();
+                            ircommand.setRemote(remoteName);
+                            ircommand.setCommand(irCommandName);
+
+                            IrCommand thingCompatibleCommand = new IrCommand();
+                            thingCompatibleCommand
+                                    .setRemote((String) channelConfiguration.get(IRtransBindingConstants.REMOTE));
+                            thingCompatibleCommand
+                                    .setCommand((String) channelConfiguration.get(IRtransBindingConstants.COMMAND));
+
+                            if (ircommand.matches(thingCompatibleCommand)) {
+                                if (sendIRcommand(ircommand,
+                                        Led.get((String) channelConfiguration.get(IRtransBindingConstants.LED)))) {
+                                    logger.debug("Sent a matching infrared command '{}' for channel '{}'", command,
+                                            channelUID);
+                                } else {
+                                    logger.warn(
+                                            "An error occurred whilst sending the infrared command '{}' for Channel '{}'",
+                                            command, channelUID);
+                                }
+                            }
+
+                        }
+                    }
+
+                    if (channel.getAcceptedItemType().equals(IRtransBindingConstants.RECEIVER_CHANNEL_TYPE)) {
+                        logger.warn("Receivers can only receive infrared commands, not send them");
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IRtrans Ethernet handler.");
+
+        // register ourselves as a Transceiver Status Listener
+        registerTransceiverStatusListener(this);
+
+        try {
+            selector = Selector.open();
+        } catch (IOException e) {
+            logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
```

Comment:
do you want to continue nonetheless or should you set the status to OFFLINE and return?